### PR TITLE
feat: update `sic_classification` to match the 1987 revision of the US SIC codes

### LINKED ISSUE #356: bug: `sic_classification` doesn't seem to be consistent with current SIC codes online
The codes in the exported `sic_classification` dataset require 5-digit codes:
Screenshot 2024-03-19 at 09 14 16

The SIC code database only allows codes containing a maximum of 4-digits: 
Screenshot 2024-03-19 at 09 14 50

It is very likely that the `sic_classification` dataset is outdated, as it was first merged 4 years ago. 

Possible actions (not mutually exclusive):
Determine if the dataset is referring to some different SIC classification system
Deprecate the sic_classification dataset 
Update the sic_classification dataset to the latest version 

[AB#10292](https://dev.azure.com/RMI-PACTA/d77fc8f8-abac-42ae-9e31-846f9901290d/_workitems/edit/10292)

### PR DESCRIPTION
**Pull request by jdhoffa**
As documented here: https://www.osha.gov/data/sic-search

Note: it seems that the original `sic_classification` bridge was referencing the South African SIC standard, as documented here: 
https://www.statssa.gov.za/additional_services/sic/mdvdvmg4.htm#MAJOR%20DIVISIONS,DIVISIONS%20AND%20MAJOR%20GROUPS

We likely should assess if we want to continue to support that dataset (and probably need to name it something more specific to identify what it refers to).

Maybe closes #356

### CODE REVIEW

**Comment by jdhoffa** [comment]
Based on the decision here: https://dev.azure.com/RMI-PACTA/2DegreesInvesting/_workitems/edit/10292#30590656
We will update the `sic_classification` dataset to the US SIC codes for clarity. 

If removing the SA SIC codes causes any issues, it is expected that the functionality soon to be introduced by https://github.com/RMI-PACTA/r2dii.match/issues/447 
will handle this. 

In particular, users will be able to flag that classification systems are missing. 

@jacobvjk This is ready for review

Closes #356 and #344

**Comment by jacobvjk** [comment]
Given that we are using a 1987 revision here, I am wondering if SIC was largely superseded by other systems in the US and only the name persisted in other jurisdictions. Not for this PR, but definitely something to consider

**Comment by jdhoffa** [comment]
> Given that we are using a 1987 revision here, I am wondering if SIC was largely superseded by other systems in the US and only the name persisted in other jurisdictions. Not for this PR, but definitely something to consider

Probably!

File: data-raw/classification_bridge.R [hunk 1]
```diff
@@ -140,16 +140,49 @@ naics_classification <- dplyr::select(
...
+  sector = dplyr::case_when(
+    grepl("^5012", SIC) ~ "automotive",
+    grepl("^451", SIC) ~ "aviation",
+    grepl("^452", SIC) ~ "aviation",
+    grepl("^324", SIC) ~ "cement",
+    grepl("^122", SIC) ~ "coal",
+    grepl("^124", SIC) ~ "coal",
+    grepl("^13", SIC) ~ "oil and gas",
+    grepl("^4911", SIC) ~ "power",
+    grepl("^331", SIC) ~ "steel",
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:162]:
```suggestion
    SIC %in% c("3312", "3315", "3316", "3317", "3324", "3325", "3462") ~ "steel",
```

Comment (jacobvjk) [reply data-raw/classification_bridge.R:162]: 3313 is unfortunately out of scope, foundries are in scope

File: data-raw/classification_bridge.R [hunk 2]
```diff
@@ -140,16 +140,49 @@ naics_classification <- dplyr::select(
...
+    TRUE ~ "not in scope"
+  ),
+  borderline = dplyr::case_when(
+    grepl("^452", SIC) ~ TRUE,
+    grepl("^124", SIC) ~ TRUE,
+    SIC %in% c("1321", "1381", "1382", "1389") ~ TRUE,
+    SIC %in% c("3313", "3315", "3316", "3317") ~ TRUE,
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:169]:
```suggestion
    SIC %in% c("3315", "3316", "3317", "3462") ~ TRUE,
```

Comment (jacobvjk) [reply data-raw/classification_bridge.R:169]: 3313 is not in scope

File: data-raw/classification_bridge.R [hunk 3]
```diff
@@ -140,16 +140,49 @@ naics_classification <- dplyr::select(
...
+  col_types = "ccccc"
+  )
+
+# freeze the data in this package in-case the above link ever gets removed
+# readr::write_csv(sic_classification_raw, "data-raw/sic_classification.csv")
+
+sic_classification <- dplyr::mutate(
+  sic_classification_raw,
+  sector = dplyr::case_when(
+    grepl("^5012", SIC) ~ "automotive",
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:154]:
```suggestion
    grepl("^371", SIC) ~ "automotive",
    SIC %in% c("5012", "5013") ~ "automotive",
```

File: data-raw/classification_bridge.R [hunk 4]
```diff
@@ -140,16 +140,49 @@ naics_classification <- dplyr::select(
...
+    grepl("^324", SIC) ~ "cement",
+    grepl("^122", SIC) ~ "coal",
+    grepl("^124", SIC) ~ "coal",
+    grepl("^13", SIC) ~ "oil and gas",
+    grepl("^4911", SIC) ~ "power",
+    grepl("^331", SIC) ~ "steel",
+    TRUE ~ "not in scope"
+  ),
+  borderline = dplyr::case_when(
+    grepl("^452", SIC) ~ TRUE,
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:166]:
```suggestion
    grepl("^371", SIC) ~ TRUE,
    grepl("^5013", SIC) ~ TRUE,
    grepl("^451", SIC) ~ TRUE,
    grepl("^452", SIC) ~ TRUE,
```

File: data-raw/classification_bridge.R [hunk 5]
```diff
@@ -140,16 +140,49 @@ naics_classification <- dplyr::select(
...
+    grepl("^5012", SIC) ~ "automotive",
+    grepl("^451", SIC) ~ "aviation",
+    grepl("^452", SIC) ~ "aviation",
+    grepl("^324", SIC) ~ "cement",
+    grepl("^122", SIC) ~ "coal",
+    grepl("^124", SIC) ~ "coal",
+    grepl("^13", SIC) ~ "oil and gas",
+    grepl("^4911", SIC) ~ "power",
+    grepl("^331", SIC) ~ "steel",
+    TRUE ~ "not in scope"
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:163]:
```suggestion
    grepl("^44", SIC) ~ "shipping",
    TRUE ~ "not in scope"
```

File: data-raw/classification_bridge.R [hunk 6]
```diff
@@ -140,16 +140,49 @@ naics_classification <- dplyr::select(
...
+  ),
+  borderline = dplyr::case_when(
+    grepl("^452", SIC) ~ TRUE,
+    grepl("^124", SIC) ~ TRUE,
+    SIC %in% c("1321", "1381", "1382", "1389") ~ TRUE,
+    SIC %in% c("3313", "3315", "3316", "3317") ~ TRUE,
+    TRUE ~ FALSE
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:170]:
```suggestion
    SIC %in% c("4911", "4931") ~ TRUE,
    SIC %in% c("4482", "4491", "4492", "4493", "4499") ~ TRUE,
    TRUE ~ FALSE
```

File: data-raw/classification_bridge.R [hunk 7]
```diff
@@ -140,16 +140,49 @@ naics_classification <- dplyr::select(
...
+  sic_classification_raw,
+  sector = dplyr::case_when(
+    grepl("^5012", SIC) ~ "automotive",
+    grepl("^451", SIC) ~ "aviation",
+    grepl("^452", SIC) ~ "aviation",
+    grepl("^324", SIC) ~ "cement",
+    grepl("^122", SIC) ~ "coal",
+    grepl("^124", SIC) ~ "coal",
+    grepl("^13", SIC) ~ "oil and gas",
+    grepl("^4911", SIC) ~ "power",
```
Comment (jacobvjk) [inline data-raw/classification_bridge.R:161]:
```suggestion
    SIC %in% c("4911", "4931") ~ "power",
```

**Review by jacobvjk — CHANGES_REQUESTED** [review]
minor adjustments

**Review by jacobvjk — APPROVED** [review]
lgtm

**Pull request merged**